src/Shortcodes/FileShortcodeProvider.php

```diff
@@ -136,17 +136,30 @@ public static function handle_shortcode($arguments, $content, $parser, $shortcod
      */
     protected static function getCachedMarkup($cache, $cacheKey, $arguments): string
     {
+        // Retrieve the cache markup
         $item = $cache->get($cacheKey);
+        if (empty($item) || empty($item['markup'])) {
+            // This is a cache miss
+            return '';
+        }
+
+        // Check the file exists
+        $filename = $item['filename'];
+        $hash = $item['hash'];
         $assetStore = Injector::inst()->get(AssetStore::class);
-        if ($item && $item['markup'] && !empty($item['filename'])) {
-            // Initiate a protected asset grant if necessary
-            $allowSessionGrant = static::getGrant(null, $arguments);
-            if ($allowSessionGrant && $assetStore->exists($item['filename'], $item['hash'])) {
-                $assetStore->grant($item['filename'], $item['hash']);
-                return $item['markup'];
-            }
+        if (empty($filename) || empty($hash) || !$assetStore->exists($filename, $hash)) {
+            // Cache entry is wrong, delete it
+            $cache->delete($cacheKey);
+            return '';
         }
-        return '';
+
+        // Shortcode can be configured to session grant access to files even if the file is protected
+        $allowSessionGrant = static::getGrant(null, $arguments);
+        if ($allowSessionGrant) {
+            $assetStore->grant($filename, $hash);
+        }
```

Comment:
```suggestion
        if (static::getGrant(null, $arguments)) {
            $assetStore->grant($filename, $hash);
        }
```
We don't need that variable. We never use it outside of this condition anyway.

Reply:
Done.